src/Std/Data/DHashMap/Internal/RawLemmas.lean

```diff
@@ -1722,4 +1725,561 @@ end Const
+    m.1.size - 1 ≤ (Const.alter m k f).1.size := by
+  simp [size_alter, h]
+  split <;> try split
+  all_goals omega
+
+theorem get?_alter (h : m.1.WF) {k k' : α} {f : Option β → Option β} :
+    Const.get? (Const.alter m k f) k' =
+      if k == k' then
+        f (Const.get? m k)
+      else
+        Const.get? m k' := by
+  simp_to_model using List.Const.getValue?_alterKey
+
+theorem get_alter (h : m.1.WF) {k k' : α} {f : Option β → Option β}
+    (hc : (Const.alter m k f).contains k') :
+    Const.get (Const.alter m k f) k' hc =
+      if heq : k == k' then
+        haveI h' : (f (Const.get? m k)).isSome := by rwa [contains_alter _ h, if_pos heq] at hc
+        (f (Const.get? m k)).get <| h'
+      else
+        haveI h' : m.contains k' := by rwa [contains_alter _ h, if_neg heq] at hc
+        Const.get m k' h' := by
+  simp_to_model using List.Const.getValue_alterKey
+
+theorem get_alter_self (h : m.1.WF) {k : α} {f : Option β → Option β}
+    {hc : (Const.alter m k f).contains k} :
+    haveI h' : (f (Const.get? m k)).isSome := by rwa [contains_alter _ h, BEq.refl] at hc
+    Const.get (Const.alter m k f) k hc = (f (Const.get? m k)).get h' := by
+  simp_to_model using List.Const.getValue_alterKey_self
+
+theorem get!_alter {k k' : α} (h : m.1.WF) [Inhabited β] {f : Option β → Option β} :
+    Const.get! (Const.alter m k f) k' =
+      if k == k' then
+        (f (Const.get? m k)).get!
+      else
+        Const.get! m k' := by
+  simp_to_model using List.Const.getValue!_alterKey
+
+theorem getD_alter {k k' : α} {v : β} (h : m.1.WF) {f : Option β → Option β} :
+    Const.getD (Const.alter m k f) k' v =
+      if k == k' then
+        f (Const.get? m k) |>.getD v
+      else
+        Const.getD m k' v := by
+  simp_to_model using List.Const.getValueD_alterKey
+
+theorem getD_alter_self {k : α} {v : β} (h : m.1.WF) {f : Option β → Option β} :
+    Const.getD (Const.alter m k f) k v = (f (Const.get? m k)).getD v := by
+  simp only [h, getD_alter, BEq.refl, reduceIte]
+
+theorem getKey?_alter (h : m.1.WF) {k k' : α} {f : Option β → Option β} :
+    (Const.alter m k f).getKey? k' =
+      if k == k' then
+        if (f (Const.get? m k)).isSome then some k else none
+      else
+        m.getKey? k' := by
+  simp_to_model using List.Const.getKey?_alterKey
+
+theorem getKey!_alter [Inhabited α] {k k' : α} (h : m.1.WF) {f : Option β → Option β} :
+    (Const.alter m k f).getKey! k' =
+      if k == k' then
+        if (f (Const.get? m k)).isSome then k else panic ""
+      else
+        m.getKey! k' := by
+  simp_to_model using List.Const.getKey!_alterKey
+
+theorem getKey_alter [Inhabited α] {k k' : α} (h : m.1.WF) {f : Option β → Option β}
+    (hc : (Const.alter m k f).contains k') :
+    (Const.alter m k f).getKey k' hc =
+      if heq : k == k' then
+        k
+      else
+        haveI h' : m.contains k' := by rwa [contains_alter _ h, if_neg heq] at hc
+        m.getKey k' h' := by
+  simp_to_model using List.Const.getKey_alterKey
```

Comment:
I do find it a bit strange that you proved `getKey_alter` in this file, but `Const.getKey_alter` in `List/Associative.lean`. As discussed, I don't mind either way, but I think it should be consistent.